modules/gapi/src/compiler/gcompiler.cpp

```diff
@@ -317,14 +317,24 @@ void cv::gimpl::GCompiler::validateInputMeta()
                      "got " + std::to_string(m_metas.size()) + " meta arguments)"));
     }
 
-    const auto meta_matches = [](const GMetaArg &meta, const GProtoArg &proto) {
+    const auto meta_matches = [](const GMetaArg &meta, const GProtoArg &proto, std::ostream& tracer) {
         switch (proto.index())
         {
         // FIXME: Auto-generate methods like this from traits:
         case GProtoArg::index_of<cv::GMat>():
         case GProtoArg::index_of<cv::GMatP>():
-            return util::holds_alternative<cv::GMatDesc>(meta);
-
+        {
+            bool is_valid = util::holds_alternative<cv::GMatDesc>(meta);
+            if (is_valid)
+            {
+                if(cv::empty_gmat_desc() == cv::util::get<cv::GMatDesc>(meta))
+                {
+                    is_valid = false;
+                    tracer << "empty cv::Mat is not allowed as compile argument";
```

Comment:
Empty cv::Mat (as the cv::Mat in general) is not a compile argument, so this message may be misleading

Reply:
Also - does this check handle the case when an empty `cv::Mat` is passed to `.apply()` ?

Reply:
It is more correct but again, we don't deal with `cv::Mat` at all when compiling a graph, so referring to a structure which may not exist at the time can be misleading.